magicgui/widgets/__init__.py

```diff
@@ -0,0 +1,54 @@
+"""The main (instantiatable) widgets from magicgui.
+
+While the primary API is to use the `@magicgui.magicgui` decorator, one can
+also instantiate widgets directly using any of these objects.
+
+The `create_widget()` function is a helper function that will instantiate and
+return the appropriate widget subclass given the arguments (type, annotation)
+to the function.
+"""
+
+from ._bases import Widget, create_widget
+from ._concrete import (
+    CheckBox,
+    ComboBox,
+    Container,
+    DateTimeEdit,
+    FileEdit,
+    FloatSlider,
+    FloatSpinBox,
+    Label,
+    LineEdit,
+    LiteralEvalLineEdit,
+    LogSlider,
+    PushButton,
+    RadioButton,
+    RangeEdit,
+    SliceEdit,
+    Slider,
+    SpinBox,
+    TextEdit,
+)
+
+__all__ = [
+    "CheckBox",
+    "ComboBox",
+    "Container",
+    "create_widget",
+    "DateTimeEdit",
+    "FileEdit",
+    "FloatSlider",
+    "FloatSpinBox",
+    "Label",
+    "LineEdit",
+    "LiteralEvalLineEdit",
+    "LogSlider",
+    "PushButton",
+    "RadioButton",
+    "RangeEdit",
+    "SliceEdit",
+    "Slider",
+    "SpinBox",
+    "TextEdit",
+    "Widget",
+]
```

Comment:
Am I right in thinking this is the core list of widgets we support and their names?

I'm taking a look now at other control widgets provided in other ecosystems like `Shiny` https://shiny.rstudio.com/tutorial/written-tutorial/lesson3/ and `ipywidgets` https://ipywidgets.readthedocs.io/en/latest/examples/Widget%20List.html and it looks like we have good coverage with our list.

Do you think there are benefits that could come from standardizing our API with the ipywidgets case, both at the level of the widgets we provide and their input parameters?

With the new 0.2.0 architecture could one in fact one day imagine an HTML backend supported by ipywidgets, no? Or alternatively if we could match APIs, could we think about some of this effort as now providing a Qt backend to the ipywidgets interface along with magic to go from a function definition to that widgets interface?

Note that I'm absolutely fine to merge this PR as is without change, but curious how you think about this, and maybe this could be a direction the library could move.

Reply:
> Do you think there are benefits that could come from standardizing our API with the ipywidgets case, both at the level of the widgets we provide and their input parameters?

yes, and now would be a good time to do it.  Definitely might help to keep the naming consistent... not immediately sure about the parameters they each accept, but if it works out well, why not!

> With the new 0.2.0 architecture could one in fact one day imagine an HTML backend supported by ipywidgets, no? 

def.  as mentioned elsewhere, I was already starting to look into this.  i think it should be doable (if not a bit overlapping in functionality with the `ipywidgets.interact` function).

> Note that I'm absolutely fine to merge this PR as is without change, but curious how you think about this, and maybe this could be a direction the library could move.

I think we should rename widgets here as desired, and then tackle widget options later.

Reply:
> I think we should rename widgets here as desired, and then tackle widget options later.

This sounds like a reasonable step (but not required). I'll post a longer response in the main thread, very exciting though!!